contracts/lifecycle/TimeFrame.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity ^0.5.2;
+
+/**
+ * @title Provides generic functionality for a TimeFrame.
+ * @author Sam Ruberti <sam@GRAFFT.io>
+ * @dev This library provides useful methods for governing a start and an end time.
+ *
+ * WARNING: This should NOT be used for contracts that depend on time for an outcome where a miner can benefit.
+ * One example would be a lottery where the first submission of an answer to a problem is rewarded. In this case the
+ * contract would be vulnerable to malicious miners who could read the answer of a not-yet-mined transaction then
+ * substitute their own answer in a new transaction before the victim's transaction, stealing the reward in the process.
+ *
+ * SAFE USE: TimeFrame CAN be used for approximate times that govern periods when methods and actions may occur.
+ * An example is an escrow that contains a start and expiration date. You can read about timestamp security
+ * vulnerabilities here:
+ * https://github.com/ethereumbook/ethereumbook/blob/first_edition_first_print/09smart-contracts-security.asciidoc#block-timestamp-manipulation
+ *
+ * IMPORTANT I emphasize that the time windows is not too narrow (e.g. a couple of seconds)
+ */
+
+
+/* solhint-disable not-rely-on-time*/
+library TimeFrame {
+    struct Epoch {
+        uint256 start;
```

Comment:
I'd rename these to `_start` and `_end` to emphasize the fact that they should not be used directly (and also maybe add a comment explaining this). There's a bit in our [API stability](https://forum.zeppelin.solutions/t/api-stability/138) document that talks about this:

>* Libraries
>Some of our Solidity libraries use structs to handle internal data that should not be accessed directly (e.g. Roles). There’s an open issue in the Solidity repository requesting a language feature to prevent said access, but it looks like it won’t be implemented any time soon. Because of this, we will use leading underscores and mark said structs to make it clear to the user that its contents and layout are not part of the API.

Reply:
I added an underscore to all internal components